backend/src/utils/auth.ts

```diff
@@ -0,0 +1,122 @@
+import { RequestHandler, Request, Response, Router, NextFunction } from 'express';
+import admin from 'firebase-admin';
+import { HandlerError } from './errors';
+import PermissionsManager from './permissionsManager';
+import { app as adminApp, env } from '../firebase';
+import MembersDao from '../dao/MembersDao';
+import rbacConfig from '../../rbac.json';
+import { AuthRole } from '../types/DataTypes';
+import AuthRoleDao from '../dao/AuthRoleDao';
+
+const getUserEmailFromRequest = async (request: Request): Promise<string | undefined> => {
+  const idToken = request.headers['auth-token'];
+  if (typeof idToken !== 'string') return undefined;
+  const decodedToken = await admin.auth(adminApp).verifyIdToken(idToken);
+  return decodedToken.email;
+};
+
+const getUserRole = async (user: IdolMember): Promise<AuthRole | undefined> => {
+  const authRoleDao = new AuthRoleDao();
+  return authRoleDao.getAuthRole(user);
+};
+
+const loginCheckedHandler =
+  (handler: (req: Request, user: IdolMember) => Promise<Record<string, unknown>>): RequestHandler =>
+  async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals;
+    try {
+      res.status(200).send(await handler(req, user));
+    } catch (error) {
+      if (error instanceof HandlerError) {
+        res.status(error.errorCode).send({ error: error.reason });
+        return;
+      }
+      res.status(500).send({ error: `Failed to handle the request due to ${error}.` });
+    }
+  };
+
+const isAuthorized = async (
+  req: Request,
+  resource: string,
+  user: IdolMember,
+  rbacConfig: any
+): Promise<boolean> => {
+  const userRole = await getUserRole(user);
+  const resourceRBACConfig = rbacConfig.resources[resource];
+
+  if (userRole === 'admin') return true;
+
+  if (req.method === 'GET') {
```

Comment:
I see you have another PR out to make our endpoints more RESTful. Hopefully all our endpoints follow the pattern of "GET" is for read-only, and everything else involves some kind of write operation.